service/dap/server.go

```diff
@@ -1534,6 +1542,16 @@ func (s *Server) onStackTraceRequest(request *dap.StackTraceRequest) {
 			stackFrames[i].Source = dap.Source{Name: filepath.Base(clientPath), Path: clientPath}
 		}
 		stackFrames[i].Column = 0
+
+		g, err := s.debugger.FindGoroutine(goroutineID)
+		if err == nil {
+			userLoc := g.UserCurrent()
+			userLocPackageName := fnPackageName(&userLoc)
+			packageName := fnPackageName(loc)
+			if userLocPackageName != "runtime" && packageName == "runtime" {
```

Comment:
Perhaps this deserves a helper? Also, how does this relate to System() from #2504?

Reply:
This is trying to extract that same information. I can leave a TODO to use the check from that change.